odoo/models.py

```diff
@@ -2188,6 +2188,8 @@ def _read_group_format_result(self, data, annotated_groupbys, groupby, domain):
             if value:
                 if ftype == 'many2one':
                     value = value[0]
+                elif ftype == 'many2many':
+                    d = [(gb['field'], 'in', (value,))]
```

Comment:
the problem with this is that `expression.py` will log a warning saying that you should really use 'in' when using relational fields, but it does work all the same (I could drop the warning if you think it's ok).

also while we're at it, I think the domain format definition could be delegated to the field instead of having this ugly branching here, but that's something for the read_group refactoring I guess

Reply:
Ok so the warning I was talking about was `The domain term '('user_ids', '=', (104,))' should use the 'in' or 'not in' operator.`

When passing the `__domain` value to a `search`, what I failed to realize was that the problem was not the left hand operator (the field) but rather the operator `=` being used with a tuple `(104,)`.

So you're right, we can drop this and just use the default